<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1151</width>
    <height>647</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="openImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Open Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="showimgIn">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>20</y>
      <width>480</width>
      <height>320</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="grayAImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Gray (Avg)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="grayBImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Gray (Mental)</string>
    </property>
   </widget>
   <widget class="QLabel" name="showimgOut">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>20</y>
      <width>480</width>
      <height>320</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QChartView" name="showHis">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>350</y>
      <width>1011</width>
      <height>271</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="threshold">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>490</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="thresholdImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>460</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Threshold</string>
    </property>
   </widget>
   <widget class="QPushButton" name="brightnessImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Brightness</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="brightness">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLineEdit" name="contrast">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>420</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="contrastImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>390</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Contrast</string>
    </property>
   </widget>
   <widget class="QPushButton" name="autoContrastImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Auto Contrast</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="grayLevel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>560</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="grayLevelImg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Gray Level</string>
    </property>
   </widget>
   <widget class="QPushButton" name="grayBImg_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Compare Gray</string>
    </property>
   </widget>
   <widget class="QPushButton" name="autoContrastImg_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>250</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Zoom in 2X</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1151</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>openImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>openImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>166</x>
     <y>67</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grayAImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>grayAImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>141</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grayBImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>grayBImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>thresholdImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>thresholdImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>brightnessImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>brightnessImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>contrastImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>contrastImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>423</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoContrastImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>autoContrast()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>89</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grayLevelImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>grayQuantization()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>563</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grayBImg_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>getGrayChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoContrastImg_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>enlargeImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>74</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openImg()</slot>
  <slot>grayAImg()</slot>
  <slot>grayBImg()</slot>
  <slot>thresholdImg()</slot>
  <slot>brightnessImg()</slot>
  <slot>contrastImg()</slot>
  <slot>autoContrast()</slot>
  <slot>grayQuantization()</slot>
  <slot>getGrayChange()</slot>
  <slot>enlargeImg()</slot>
 </slots>
</ui>
